test/test_python_dispatch.py

```diff
@@ -497,9 +495,12 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         b = B(torch.empty(1))
         with self.assertRaises(ErrorA):
             a + a
-
-        # B has precedence over A due to the subclass relationship
         with self.assertRaises(ErrorB):
+            a + b
+
+        # B has precedence over A due to the subclass relationship yet
+        # modes take precedence over arguments
+        with self.assertRaises(ErrorA):
```

Comment:
Flagging that this behavior changed from the original version of torch dispatch (fka python) mode. IMO this behavior feels more like how I would expect modes to work, but just wanted to flag that it's a change in behavior

Reply:
Thanks for the flag. This change looks fine. Even better if we just get rid of direct tensor subclass mode, that will eliminate the question here entirely.